test/integration/category_builder_test.go

```diff
@@ -0,0 +1,177 @@
+)
+
+var expectedGroups = map[string]bool{
+	"apiregistration.k8s.io/v1beta1":       false,
+	"extensions/v1beta1":                   false,
+	"apps/v1":                              false,
+	"apps/v1beta1":                         false,
+	"apps/v1beta2":                         false,
+	"events.k8s.io/v1beta1":                false,
+	"authentication.k8s.io/v1":             false,
+	"authentication.k8s.io/v1beta1":        false,
+	"authorization.k8s.io/v1":              false,
+	"authorization.k8s.io/v1beta1":         false,
+	"autoscaling/v1":                       false,
+	"autoscaling/v2beta1":                  false,
+	"batch/v1":                             false,
+	"batch/v1beta1":                        false,
+	"batch/v2alpha1":                       false,
+	"certificates.k8s.io/v1beta1":          false,
+	"networking.k8s.io/v1":                 false,
+	"policy/v1beta1":                       false,
+	"authorization.openshift.io/v1":        false,
+	"rbac.authorization.k8s.io/v1":         false,
+	"rbac.authorization.k8s.io/v1beta1":    false,
+	"storage.k8s.io/v1":                    false,
+	"storage.k8s.io/v1beta1":               false,
+	"admissionregistration.k8s.io/v1beta1": false,
+	"apiextensions.k8s.io/v1beta1":         false,
+	"apps.openshift.io/v1":                 false,
+	"build.openshift.io/v1":                false,
+	"image.openshift.io/v1":                false,
+	"network.openshift.io/v1":              false,
+	"oauth.openshift.io/v1":                false,
+	"project.openshift.io/v1":              false,
+	"quota.openshift.io/v1":                false,
+	"route.openshift.io/v1":                false,
+	"security.openshift.io/v1":             false,
+	"template.openshift.io/v1":             false,
+	"user.openshift.io/v1":                 false,
+	"/v1": false,
+}
+
+var expectedCategories = map[string][]string{
+	"DaemonSet/extensions/v1beta1":                {"all"},
+	"Deployment/extensions/v1beta1":               {"all"},
+	"ReplicaSet/extensions/v1beta1":               {"all"},
+	"DaemonSet/apps/v1":                           {"all"},
+	"Deployment/apps/v1":                          {"all"},
+	"ReplicaSet/apps/v1":                          {"all"},
+	"StatefulSet/apps/v1":                         {"all"},
+	"Deployment/apps/v1beta1":                     {"all"},
+	"StatefulSet/apps/v1beta1":                    {"all"},
+	"DaemonSet/apps/v1beta2":                      {"all"},
+	"Deployment/apps/v1beta2":                     {"all"},
+	"ReplicaSet/apps/v1beta2":                     {"all"},
+	"StatefulSet/apps/v1beta2":                    {"all"},
+	"HorizontalPodAutoscaler/autoscaling/v1":      {"all"},
+	"HorizontalPodAutoscaler/autoscaling/v2beta1": {"all"},
+	"Job/batch/v1":                                {"all"},
+	"DeploymentConfig/apps.openshift.io/v1":       {"all"}, // FIX: discovered with no categories
+	"BuildConfig/build.openshift.io/v1":           {"all"}, // FIX: discovered with no categories
```

Comment:
@deads2k I believe I am removing categories from the wrong resources. I can restore the `Categories()` method in the files above, however per the results of this test, `/oapi` resources continue to be requested as part of `oc get all` despite being discovered in this test without any categories